pkg/systemd/hwinfo.jsx

```diff
@@ -65,20 +68,115 @@ const SystemInfo = ({ info }) => (
+        if (this.props.onClose)
+            this.props.onClose();
+    }
+
+    nosmtRow() {
+        /* nosmt */
+        let promises = [cockpit.spawn(["lscpu", "--json"]), cockpit.spawn(["cat", "/proc/cmdline"])];
+        Promise.all(promises).then(values => {
+            let threads_per_core = Number(JSON.parse(values[0]).lscpu.filter(value => value.field.indexOf("Thread(s) per core") !== -1)[0].data);
+            let smt_disabled = values[1].indexOf("nosmt") !== -1;
+            if (values[1].indexOf("nosmt=") === -1 && (threads_per_core > 1 || (threads_per_core === 1 && smt_disabled)))
+                this.setState({ nosmt: smt_disabled });
+        });
+    }
+
+    saveAndReboot() {
+        this.setState(state => {
+            let options = [];
+            let promises = [];
+
+            if (state.nosmt)
+                options.push(['set', 'nosmt']);
+            else if (state.nosmt !== undefined)
+                options.push(['remove', 'nosmt']);
+
+            for (let option of options)
+                promises.push(cockpit.script(kernelopt_sh, option, { "superuser": true, "err": "message" }));
+
+            Promise.all(promises)
+                    .then(() => {
+                        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: true, err: "message" });
+                    })
+                    .catch(() => {
+                        this.setState({ rebooting: false });
+                    });
+        });
+    }
+
+    render() {
+        let rows = [];
+
+        if (this.state.nosmt !== undefined)
+            rows.push((<ListingRow key="nosmt" columns={ [(<b>{ _("Disable symmetric multithreading") } (nosmt)</b>),
+                { element: <OnOffSwitch disabled={this.state.rebooting}
+                onChange={ value => this.setState({ nosmt: value }) }
+                state={ this.state.nosmt } />, tight: true }] } />));
+
+        if (rows.length === 0)
+            rows.push((<ListingRow columns={ [(<b>{ _("No mitigations available") }</b>)] } />));
+
+        return (
+            <Modal show={this.props.show} id="cpu-mitigations-dialog">
+                <Modal.Header>
+                    <Modal.Title> { _("CPU Security Toggles") } </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    { _("Software-based workarounds help prevent CPU security issues. These mitigations have the side effect of reducing performance. Turn off these settings at your own risk. read more link TODO") }
+                    <Listing>
+                        { rows }
+                    </Listing>
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button bsStyle='default' className='btn-cancel' disabled={this.state.rebooting} onClick={this.close}>
+                        {_("Cancel")}
+                    </Button>
+                    <Button bsStyle='primary' disabled={this.state.rebooting} onClick={this.saveAndReboot}>
```

Comment:
How about bsStyle="danger"?  @garrett?

Reply:
Is this for the reboot button?

We do use a danger style on the overview page, and it could possibly cause data loss (VMs & containers that are running, users signed in, etc.), so it would be consistent. :+1: